ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressInput.service.ts

```diff
@@ -222,38 +216,36 @@ export class AddressInputService implements OnDestroy {
 
     // this is the address that we are building from data in the form
     // todo: can we factor out all these distinctUntilChanged. We have them everywhere
-    const formAddress$ = componentState$.pipe(
+    const formAddress$: Observable<Address> = componentState$.pipe(
       filter((compState) => compState.formDataSource === 'COMPONENT'),
       // only care about changes
       distinctUntilChanged((x, y) => _.isEqual(x.formData, y.formData)),
       // but don't care about initial state
       skip(1),
       map(compState => this.buildAddressFromFormData(compState.formData, compState.countryInfo)),
       distinctUntilChanged((x, y) => _.isEqual(x, y)),
-      tap((newAddress) => console.debug('generated new formAddress$ %o', newAddress)),
       share());
 
     const street1Changed$: Observable<boolean> = formAddress$.pipe(
       pluck('street1'),
+      startWith(null as string | null),
       pairwise(),
       map(([prev, current]) => prev !== current),
       startWith(false),
-      tap((changed) => console.debug('generated new street1changed with value:' + changed)),
       share());
 
-
     // if street1 changed, user is actually typing a value or trying to use google autocomplete
     // since we don't know what they are up to, we will wait before we push the address
     // if we get an autocomplete google address, then we won't push the form address
     // Note that for this work properly we should always get a properly paired of address and street1Changed
-    const cancelableFormAddress$ = zip(formAddress$, street1Changed$).pipe(
-      concatMap(([formAddress, street1Changed]) =>
-        of(formAddress).pipe(
-          tap((address) => console.debug('inputs to canceLable:' + JSON.stringify(address) + street1Changed)),
-          delay(street1Changed ? 3000 : 0),
-          // If autocomplete address has come in during our delay, we don't emit the form address
-          takeUntil(this.googleAutocompleteAddress$),
-          take(1)
+    const cancelableFormAddress$ = street1Changed$.pipe(withLatestFrom(formAddress$)).pipe(
+      concatMap(([streetChanged, address]) =>
+        of(streetChanged).pipe(
+            delay(streetChanged ? 3000 : 0),
+            mapTo(address),
+            // If autocomplete address has come in during our delay, we don't emit the form address
+            takeUntil(this.googleAutocompleteAddress$),
+            take(1)
         )),
```

Comment:
Noticed that under some circumstances using `zip` was not working as I expected. Driving cancelableFormAddress from street1Changed is more direct and reliable.